.buildkite/pull_request_pipeline.yml

```diff
@@ -45,19 +47,33 @@ steps:
     retry:
       automatic:
         - limit: 3
+    env:
+      ENABLE_SONARQUBE: true
     command: |
-      set -euo pipefail
-      if [[ $BUILDKITE_PULL_REQUEST == "false" ]]; then
-        # https://github.com/elastic/logstash/pull/15486 for background
-        export ENABLE_SONARQUBE="false"
-      else
-        source .buildkite/scripts/pull-requests/sonar-env.sh
-      fi
-
+      set -euo pipefail      
       source .buildkite/scripts/common/container-agent.sh
       ci/unit_tests.sh java
     artifact_paths:
       - "**/build/test-results/javaTests/TEST-*.xml"
+      - "**/jacocoTestReport.xml"
+      - "**/build/classes/**/*.*"
+
+  - label: ":sonarqube: Continuous Code Inspection"
+    env:
+      VAULT_SONAR_TOKEN_PATH: "kv/ci-shared/platform-ingest/elastic/logstash/sonar-analyze-token"
+    agents:
+      image: "docker.elastic.co/cloud-ci/sonarqube/buildkite-scanner:latest"
+    command:
+      - "buildkite-agent artifact download --step ruby-unit-tests coverage/coverage.json ."
+      - "buildkite-agent artifact download --step java-unit-tests **/jacocoTestReport.xml ."
+      - "buildkite-agent artifact download --step java-unit-tests **/build/classes/**/*.* ."
+      - "/scan-source-code.sh"
```

Comment:
It's recommended to measure code coverage on branches(main, ...) to keep it on track and compare it to pull request coverage.

Reply:
Thanks this helps. We can add to the earlier proposed conditional (to allow if it's a pr) to also allow if branches are main or x.y.z (this is a very common conditional used across the elastic code base).

What I want to avoid is unnecessary runs/failures when we are testing pipelines on an adhoc basis using specific commits or branchs/tags outside main/8.x/7.x

Do you feel confident that it's ok to have this step always run without causing unnecessary failures or skewing results by being run from unrelated branches/commits?

Reply:
That's a good idea

Reply:
Now the sonar scan will be run on PRs and main. https://github.com/elastic/logstash/pull/15810/files#diff-044e110c951f71fa4ff9db56f0821bac0de6e1d091d779b448bd6f14e1ad1c0dR62